cli/src/diff_util.rs

```diff
@@ -749,6 +760,107 @@ pub fn show_diff_summary(
     })
 }
 
+struct DiffStat {
+    path: String,
+    added: usize,
+    removed: usize,
+}
+
+fn get_diff_stat(repo_path: String, left_content: &[u8], right_content: &[u8]) -> DiffStat {
+    let hunks = unified_diff_hunks(left_content, right_content, 0);
+    let mut adds = 0;
+    let mut dels = 0;
+    for hunk in hunks {
+        for (line_type, _content) in hunk.lines {
+            match line_type {
+                DiffLineType::Context => {}
+                DiffLineType::Removed => dels += 1,
+                DiffLineType::Added => adds += 1,
+            }
+        }
+    }
+    DiffStat {
+        path: repo_path,
+        added: adds,
+        removed: dels,
+    }
+}
+
+pub fn show_diff_stat(
+    ui: &Ui,
+    formatter: &mut dyn Formatter,
+    workspace_command: &WorkspaceCommandHelper,
+    tree_diff: TreeDiffIterator,
+) -> Result<(), CommandError> {
+    let mut stats: Vec<DiffStat> = vec![];
+    let mut max_path_length = 0;
+    let mut max_diffs = 0;
+    for (repo_path, diff) in tree_diff {
+        let path = workspace_command.format_file_path(&repo_path);
+        let mut left_content: Vec<u8> = vec![];
+        let mut right_content: Vec<u8> = vec![];
+        match diff {
+            tree::Diff::Modified(left, right) => {
+                left_content = diff_content(workspace_command.repo(), &repo_path, &left)?;
+                right_content = diff_content(workspace_command.repo(), &repo_path, &right)?;
+            }
+            tree::Diff::Added(right) => {
+                right_content = diff_content(workspace_command.repo(), &repo_path, &right)?;
+            }
+            tree::Diff::Removed(left) => {
+                left_content = diff_content(workspace_command.repo(), &repo_path, &left)?;
+            }
+        }
+        max_path_length = max(max_path_length, path.len());
+        let stat = get_diff_stat(path, &left_content, &right_content);
+        max_diffs = max(max_diffs, stat.added + stat.removed);
+        stats.push(stat);
+    }
+
+    let display_width = usize::from(ui.term_width().unwrap_or(80)) - 4; // padding
+    let max_bar_length =
+        display_width - max_path_length - " | ".len() - max_diffs.to_string().len() - 1;
+    let factor = if max_diffs < max_bar_length {
+        1.0
+    } else {
+        max_bar_length as f64 / max_diffs as f64
+    };
+    let number_padding = max_diffs.to_string().len();
+
+    formatter.with_label("diff", |formatter| {
+        let mut total_added = 0;
+        let mut total_removed = 0;
+        for stat in &stats {
+            total_added += stat.added;
+            total_removed += stat.removed;
+            let bar_added = (stat.added as f64 * factor).ceil() as usize;
+            let bar_removed = (stat.removed as f64 * factor).ceil() as usize;
+            // pad to max_path_length
+            write!(
+                formatter.labeled("stat-line"),
+                " {:<max_path_length$} | {:>number_padding$}{}",
```

Comment:
Is the leading space intentional?

Reply:
Yeah, I don't know where it comes from but if you look at `git diff --stat` it also adds the leading space. I think it comes from the original `diffstat` program that indents with one space.

```
jj ❯ git diff 2102de94 e414f3b7 | diffstat
 CHANGELOG.md                       |    4 ++
 cli/src/cli_util.rs                |   34 +++++-----------------
 cli/src/commands/mod.rs            |   51 ++++++++++++++++------------------
 cli/src/config-schema.json         |    5 +++
 cli/src/config/colors.toml         |    1
 cli/src/diff_util.rs               |   20 +++++++++----
 cli/src/merge_tools.rs             |   55 ++++++++++++++++++++----------------
 cli/tests/test_describe_command.rs |   89 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++-
 cli/tests/test_diff_command.rs     |   23 +++++++++++++++
 docs/config.md                     |    4 +-
 docs/operation-log.md              |    2 -
 lib/src/conflicts.rs               |   17 ++++-------
 lib/src/default_index_store.rs     |  115 ++++++++++++++++++++++-------------------------------------------------------
 lib/src/gitignore.rs               |    7 +---
 lib/src/local_backend.rs           |    4 --
 lib/src/repo.rs                    |    7 ++--
 lib/src/working_copy.rs            |   76 ++++++++++++++++++++++++--------------------------
 lib/src/workspace.rs               |    7 ++--
 lib/tests/test_conflicts.rs        |    8 -----
 19 files changed, 288 insertions(+), 241 deletions(-)
```

Reply:
Weird! Do you think we need to match that weirdness? I'd rather remove it if possible.

Reply:
We can drop it I guess? I've been cargo-cutting it for like a decade :-)

Reply:
ok, dropped.

Reply:
Yes, let's not waste a column until we find a reason to do it (at which point it's no longer "waste") :)